.gitignore

```diff
@@ -13,3 +13,5 @@ coverage.html
 .glide
 .scBuildImage
 .init
+.generate*
+**/zz*.go
```

Comment:
Sorry @duglin, I realize now that I assumed but didn't confirm that you agree to it.

Anyway, to recap, here are the reasons I gave on Slack yesterday, in favor of checking in generated code:

1. There's no dependency on code generators for devs (except those who want to make changes to the IDL file) and CI
2. We can build a reusable tiller SDK for 3rd parties (i.e. the original steward used that SDK)

Additional notes:

1. Accidental modification was one of the concerns with checking in vendored code
1. Generated code is not the same as vendored code. Its source and output are isolated to this repository
    1. If someone accidentally modifies generated code and reviewers don't catch the change, the build will fail next time code is generated. But most importantly, when the build fails, a simple diff in our codebase (not in a dependency) shows why. 
1. I must repeat the sentiment that it's important that we follow the general structure of kubernetes/kubernetes for now as best we can. This is to make it more friendly to core developers who may want to contribute to, extend, or otherwise adopt our codebase -- a goal of all of ours